### ISSUE TITLE
Rule Suggestion: disallow explicit boolean comparisons

### ISSUE THREAD

**Issue by chrismbarr** [palantir/tslint#1910]
Novice programmers might have something like this
```ts
let foo = true;
// other code...

if (foo === true) {
  //do something...
}
```

or maybe 

```ts
if (foo !== false) {
  //do something...
}
```
(I have seen both of these in real code!)

When `foo` is a boolean, this is unnecessary and could result in a linting violation.  However if `foo` can be something else like `foo: true | number | null` then these explicit checks should be allowed.

perhaps this could be added as an option to the existing [`strict-boolean-expressions`](https://palantir.github.io/tslint/rules/strict-boolean-expressions/) rule?
